crates/next-core/src/next_client_chunks/with_chunks.rs

```diff
@@ -49,16 +45,35 @@ impl Asset for WithChunksAsset {
     }
 
     #[turbo_tasks::function]
-    fn references(&self) -> AssetReferencesVc {
-        unimplemented!()
+    async fn references(&self) -> Result<AssetReferencesVc> {
+        Ok(AssetReferencesVc::cell(vec![ChunkGroupReferenceVc::new(
+            ChunkGroupVc::from_asset(
+                self.asset.into(),
+                self.chunking_context,
+                None,
+                Some(self.asset.into()),
+            ),
+        )
+        .into()]))
     }
 }
 
 #[turbo_tasks::value_impl]
 impl ChunkableAsset for WithChunksAsset {
     #[turbo_tasks::function]
-    fn as_chunk(self_vc: WithChunksAssetVc, context: ChunkingContextVc) -> ChunkVc {
-        EcmascriptChunkVc::new(context, self_vc.as_ecmascript_chunk_placeable()).into()
+    fn as_chunk(
+        self_vc: WithChunksAssetVc,
+        context: ChunkingContextVc,
+        available_assets: Option<AvailableAssetsVc>,
+        current_availability_root: Option<AssetVc>,
```

Comment:
Feels like these two are almost always passed together. Can we make them part of a single `struct Availability`?